app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/AddressLookupResult.scala

```diff
@@ -20,6 +20,7 @@ import play.api.libs.json.{Format, Json}
 
 final case class AddressLookupResult(
   postcode: Postcode,
+  filter: Option[String],
   addresses: List[Address]
```

Comment:
why does filter need to be in the AddressLookupResult? The result should just contain the address or nothing.

Reply:
it might be being used to prepopulate the form is a user goes back, and also on the form submit endpoint to check if someone's already we've got a search result for the parameters the user's entered. In that case we can skip a call to address lookup and just use the cached address lookup result.

Reply:
It's useful to hold in session so that the exact search that was used to produce the results can be replayed.